lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1697,12 +1831,29 @@ func (d *zfs) MountVolume(vol Volume, op *operations.Operation) error {
 
 			revert.Add(func() { _ = d.setDatasetProperties(dataset, "volmode=none") })
 
-			// Wait half a second to give udev a chance to kick in.
-			time.Sleep(500 * time.Millisecond)
+			// Wait a second to give udev a chance to kick in.
+			time.Sleep(time.Second)
 
 			d.logger.Debug("Activated ZFS volume", logger.Ctx{"dev": dataset})
 		}
 
+		if vol.contentType != ContentTypeBlock && d.isBlockBacking(vol) && !filesystem.IsMountPoint(mountPath) {
+			err = vol.EnsureMountPath()
```

Comment:
I would do this after `d.GetVolumeDiskPath(vol)` so if it fails you don't leave the mount path around.